src/transformers/models/vit/convert_vit_timm_to_pytorch.py

```diff
@@ -0,0 +1,268 @@
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".norm1.bias", "vit.encoder.layer." + str(i) + ".layernorm_before.bias")
+        )
+        rename_keys.append(
+            (
+                "blocks." + str(i) + ".attn.proj.weight",
+                "vit.encoder.layer." + str(i) + ".attention.output.dense.weight",
+            )
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".attn.proj.bias", "vit.encoder.layer." + str(i) + ".attention.output.dense.bias")
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".norm2.weight", "vit.encoder.layer." + str(i) + ".layernorm_after.weight")
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".norm2.bias", "vit.encoder.layer." + str(i) + ".layernorm_after.bias")
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".mlp.fc1.weight", "vit.encoder.layer." + str(i) + ".intermediate.dense.weight")
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".mlp.fc1.bias", "vit.encoder.layer." + str(i) + ".intermediate.dense.bias")
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".mlp.fc2.weight", "vit.encoder.layer." + str(i) + ".output.dense.weight")
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".mlp.fc2.bias", "vit.encoder.layer." + str(i) + ".output.dense.bias")
+        )
+
+    # projection layer + position embeddings
+    rename_keys.extend(
+        [
+            ("cls_token", "vit.embeddings.cls_token"),
+            ("patch_embed.proj.weight", "vit.embeddings.patch_embeddings.projection.weight"),
+            ("patch_embed.proj.bias", "vit.embeddings.patch_embeddings.projection.bias"),
+            ("pos_embed", "vit.embeddings.position_embeddings"),
+        ]
+    )
+
+    # pooler
+    if config.use_pooler:
+        rename_keys.extend(
+            [
+                ("pre_logits.fc.weight", "pooler.dense.weight"),
+                ("pre_logits.fc.bias", "pooler.dense.bias"),
+            ]
+        )
+
+    # classification head
+    rename_keys.extend(
+        [
+            ("head.weight", "classifier.weight"),
+            ("head.bias", "classifier.bias"),
+            ("norm.weight", "layernorm.weight"),
+            ("norm.bias", "layernorm.bias"),
+        ]
+    )
+
+    # to do: add base model support
+    # if just the base model, we should remove "vit" from all keys
+    if base_model:
+        pass
+
+    return rename_keys
+
+
+# we split up the matrix of each encoder layer into queries, keys and values
+def read_in_q_k_v(state_dict, config, base_model=False):
+    for i in range(config.num_hidden_layers):
+        # read in weights + bias of input projection layer (in timm, this is a single matrix + bias)
+        in_proj_weight = state_dict.pop("blocks." + str(i) + ".attn.qkv.weight")
```

Comment:
Same here and in the lines below, a f-string would be more readable.

Reply:
Please do not resolve a comment without treating it. This line and the line below still need to use f-strings.

Reply:
Ok, sorry I've missed these two